src/cdk/datetime/datepicker.spec.ts

```diff
@@ -0,0 +1,646 @@
+import {
+  dispatchFakeEvent,
+  DEC,
+  JAN,
+  JUL,
+  JUN,
+} from '@angular/cdk/testing';
+import {NativeDateModule} from '@angular/cdk/datetime';
+import {Component, FactoryProvider, Type, ValueProvider, ViewChild} from '@angular/core';
+import {ComponentFixture, fakeAsync, flush, TestBed} from '@angular/core/testing';
+import {FormControl, FormsModule, ReactiveFormsModule} from '@angular/forms';
+import {By} from '@angular/platform-browser';
+import {NoopAnimationsModule} from '@angular/platform-browser/animations';
+import {CdkDatepicker} from './datepicker';
+import {CdkDatepickerInput} from './datepicker-input';
+import {CdkDatepickerModule} from './index';
+import {BrowserDynamicTestingModule} from '@angular/platform-browser-dynamic/testing';
+
+describe('CdkDatepicker', () => {
+  // Creates a test component fixture.
+  function createComponent(
+      component: Type<any>,
+      imports: Type<any>[] = [],
+      providers: (FactoryProvider | ValueProvider)[] = [],
+      entryComponents: Type<any>[] = []): ComponentFixture<any> {
+
+    TestBed.configureTestingModule({
+      imports: [
+        FormsModule,
+        CdkDatepickerModule,
+        NoopAnimationsModule,
+        ReactiveFormsModule,
+        ...imports
+      ],
+      providers,
+      declarations: [component, ...entryComponents],
+    });
+
+    TestBed.overrideModule(BrowserDynamicTestingModule, {
+      set: {
+        entryComponents: [entryComponents]
+      }
+    }).compileComponents();
```

Comment:
`compileComponents` actually completes asynchronously, so you can't call `createComponent` immediately after it. `configureTestingModule` and `compileComponents` should be called separately inside of an `async` block. The tests for radio-button are a good example of this pattern.

Reply:
The `radio.spec.ts` pattern isn't within a `createComponent` function though, which I think is needed in our case because of: https://sourcegraph.com/github.com/angular/material2/-/commit/3f589cd2bf221517958eea002c2c4ee71ef769c7
(also I think we switched to `fakeAsync` instead of `async`). `src/cdk/table/table.spec.ts` (and multiple other components) implement the `createComponent` immediately after `compileComponents` within their modified version of the `createComponent` function, similar to what we are doing here.